src/main/java/io/github/thebusybiscuit/exoticgarden/schematics/org/jnbt/NBTOutputStream.java

```diff
@@ -179,8 +179,8 @@ private void writeListTagPayload(ListTag tag) throws IOException {
 
 		os.writeByte(NBTUtils.getTypeCode(clazz));
 		os.writeInt(size);
-		for(int i = 0; i < size; i++) {
-			writeTagPayload(tags.get(i));
+		for (Tag value : tags) {
+			writeTagPayload(value);
```

Comment:
That's JNBT code right here and not from ExoticGarden, nothing really speaks against modifying it but just so you are aware.
Should probably look into whether we can shade that library too anyway.